src/main/java/io/jenkins/plugins/ml/Server.java

```diff
@@ -74,26 +83,78 @@ public long getMaxResults() {
+            if( Util.fixEmptyAndTrim(serverName) == null){
+                return FormValidation.warning("* Required ");
+            }
+            String regex = "^[a-zA-Z0-9_]+$";
+            Pattern pattern = Pattern.compile(regex);
+            if( pattern.matcher(serverName).matches() ){
+                return FormValidation.ok();
+            }else{
+                return FormValidation.warning(" Try a another name. Not supported",serverName);
+            }
+        }
+
+        public FormValidation doCheckServerAddress(@QueryParameter String serverAddress) {
+            if( Util.fixEmptyAndTrim(serverAddress) == null){
+                return FormValidation.warning("* Required ");
+            }
+            else if(InetAddresses.isInetAddress(serverAddress)){
+                return FormValidation.ok();
+            }else{
+                return FormValidation.error("Malformed IP address ", serverAddress);
+            }
+        }
+
+        public FormValidation doCheckLaunchTimeout(@QueryParameter String launchTimeout) {
+           try{
+               Integer num = Integer.valueOf(launchTimeout);
+               if(num >= 0){
+                   return FormValidation.ok();
+               }
+           }catch (Exception e){
+               return FormValidation.error("Timeout should be a valid number ");
+           }
+           return FormValidation.ok();
+        }
+
+        public FormValidation doCheckMaxResults(@QueryParameter String maxResults) {
+            try{
+                Integer num = Integer.valueOf(maxResults);
+                if(num >= 1){
+                    return FormValidation.ok();
+                }
+            }catch (Exception e){
+                return FormValidation.error("Max results should be a valid number ");
+            }
+            return FormValidation.ok();
+        }
     }
 
     /**
      * Gets the {@link Server} associated with the given job.
      *
-     * @return local host
-     *         if no such was found.
+     * @param p job
+     * @return the Server configured for the job
+     * @throws IllegalStateException throws if there is either null job or no server
+     *
      */
     public static Server get(Job<?, ?> p) {
         if(p != null) {
             ServerJobProperty ijp = p.getProperty(ServerJobProperty.class);
             if (ijp != null) {
                 // Looks in global configuration for the server configured
-                Server site = ijp.getSite();
-                if (site != null) {
-                    return site;
+                Server server = ijp.getServer();
+                if (server != null) {
+                    return server;
                 }
             }
+        }else{
+            throw new IllegalStateException("A null job is passed");
         }
-        return new Server("localhost","127.0.0.1",1000,2);
-        }
+        throw new IllegalStateException("Here's an error. No server configured ");
```

Comment:
The `else` is not necessary @Loghijiaha I think? Just

```java
public static Server get(Job<?, ?> p) {
        if(p != null) {
            // create the server from the job
            return server;
        }
        throw new RuntimeException("Invalid job, failed to create IPython server");
}
```

The message is also important. It is good to imagine what someone else will think when s/he sees "Here's an error. No server configured" in the server log. Unless that person is configuring a server, there is no easy way to know that it was with the IPython Plug-in, unless we look at the complete log stack trace.

By adding something like "Failed to create IPython Server", the user or admin should immediately suspect there is something to do with this plug-in :+1: